package firmament;

import "base/reference_desc.proto";

message TaskDescriptor {
  enum TaskState {
    CREATED = 0;
    BLOCKING = 1;
    RUNNABLE = 2;
    ASSIGNED = 3;
    RUNNING = 4;
    COMPLETED = 5;
    FAILED = 6;
    ABORTED = 7;
    DELEGATED = 8;
    UNKNOWN = 9;
  }

  required uint64 uid = 1;
  required string name = 2;
  required TaskState state = 3;
  optional string job_id = 4;
  optional uint64 index = 5;
  repeated ReferenceDescriptor dependencies = 6;
  repeated ReferenceDescriptor outputs = 7;
  required bytes binary = 8;
  repeated string args = 9;
  repeated TaskDescriptor spawned = 10;
  optional string last_location = 11;
  optional string delegated_from = 12;
  optional uint64 started = 13;
  optional uint64 relative_deadline = 14;
  optional uint64 absolute_deadline = 15;
  optional uint64 equivalence_class = 16;


}

